ionoscloud/resource_server.go

```diff
@@ -948,6 +961,21 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	}
 
+	// Labels logic for update
+	if d.HasChanges("label") {
+		ls := LabelsService{ctx: ctx, client: client}
+		oldLabelsData, newLabelsData := d.GetChange("label")
+
+		// Delete the old labels.
+		if err := ls.datacentersServersLabelsDelete(dcId, d.Id(), oldLabelsData); err != nil {
+			return diag.FromErr(err)
+		}
+
+		// Create the new labels.
+		if err := ls.datacentersServersLabelsCreate(dcId, d.Id(), newLabelsData); err != nil {
+			return diag.FromErr(err)
+		}
```

Comment:
hmm we should do a diff here between the slices, otherwise if someone just adds a new label and there are 10 existing labels, we will do 10 delete requests and 11 post requests with only one being the new label?

Reply:
@cristiGuranIonos done in c2441c02e75924fac0bba764c00928cc08878b5e